mouse/mouse.nim

```diff
@@ -42,113 +51,82 @@ when defined(windows):
             Release: int
 
     const
-        Left*: ButtonTuple = (MOUSEEVENTF_LEFTDOWN, MOUSEEVENTF_LEFTUP)
-        Right*: ButtonTuple = (MOUSEEVENTF_RIGHTDOWN, MOUSEEVENTF_RIGHTUP)
-        Midle*: ButtonTuple = (MOUSEEVENTF_MIDDLEDOWN, MOUSEEVENTF_MIDDLEUP)
+        MouseButtonToEvent: array[MouseButton, ButtonTuple] = [
+            Left: (MOUSEEVENTF_LEFTDOWN, MOUSEEVENTF_LEFTUP),
+            Right: (MOUSEEVENTF_RIGHTDOWN, MOUSEEVENTF_RIGHTUP),
+            Middle: (MOUSEEVENTF_MIDDLEDOWN, MOUSEEVENTF_MIDDLEUP)
+        ]
         X*: ButtonTuple = (MOUSEEVENTF_XDOWN, MOUSEEVENTF_XUP)
 
-        Horizontal*: int = MOUSEEVENTF_HWHEEL
-        Vertical*: int = MOUSEEVENTF_WHEEL
+        ScrollOrientationToEvent = [
+            Horizontal: MOUSEEVENTF_HWHEEL,
+            Vertical: MOUSEEVENTF_WHEEL
+        ]
 
-        Relative*: string = "relative"
-        Absolute*: string = "absolute"
 
 
+    proc click*(button: MouseButton, x = 0, y = 0) =
+        mouse_event(DWORD(MouseButtonToEvent[button].Press), DWORD(x), DWORD(y), 0, 0)
+        mouse_event(DWORD(MouseButtonToEvent[button].Release), DWORD(x), DWORD(y), 0, 0)
 
+    proc press*(button: MouseButton, x = 0, y = 0) =
+        mouse_event(DWORD(MouseButtonToEvent[button].Press), DWORD(x), DWORD(y), 0, 0)
 
-    proc click*(button: ButtonTuple, x = 0, y = 0) =
-        mouse_event(DWORD(button.Press), DWORD(x), DWORD(y), 0, 0)
-        mouse_event(DWORD(button.Release), DWORD(x), DWORD(y), 0, 0)
+    proc release*(button: MouseButton, x = 0, y = 0) =
+        mouse_event(DWORD(MouseButtonToEvent[button].Release), DWORD(x), DWORD(y), 0, 0)
 
-    proc press*(button: ButtonTuple, x = 0, y = 0) =
-        mouse_event(DWORD(button.Press), DWORD(x), DWORD(y), 0, 0)
-
-    proc release*(button: ButtonTuple, x = 0, y = 0) =
-        mouse_event(DWORD(button.Release), DWORD(x), DWORD(y), 0, 0)
-
-    proc move*(x: int32, y: int32, `type` = Relative) =
-        if `type` == "relative":
+    proc move*(x, y: int, typ = Relative) =
+        case typ
+        of Relative:
             mouse_event(0x1, DWORD(x), DWORD(y), 0, 0)
-        elif `type` == "absolute":
-            SetCursorPos(x, y)
+        of Absolute:
+            SetCursorPos(int32 x, int32 y)
 
-    proc smoothMove*(x: int, y: int, smoothingStep: float, `sleep` = 3, `type` = Absolute) = 
+    proc smoothMove*(x: int, y: int, smoothingStep: float, sleep = 3, typ = Absolute) = 
         var currPos: Point
         GetCursorPos(currPos)
 
-        if `type` == "relative":
-            var x,y = calcRelative((int(currPos.x), int(currPos.y)), (x,y))
+        var
+            x = x
+            y = y
+        if typ == Relative:
```

Comment:
Same as in `calcRelative` - you were creating new variables in the `if `type` == "relative"` branch instead of changing the existing x, y ones. I've fixed that.